src/Models/Workflow.php

```diff
@@ -89,7 +89,13 @@ public function onStepFinished($job): void
     public function onStepFailed($job, Throwable $e): void
     {
         DB::transaction(function () use ($job, $e) {
-            $this->jobs_failed++;
+            /** @var self $workflow */
+            $workflow = $this->newQuery()
+                ->whereKey($this->getKey())
+                ->lockForUpdate()
+                ->firstOrFail(['jobs_failed']);
```

Comment:
Since the key is unique, I think this should be `findOrFail` so we can avoid a `LIMIT` query.